src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker_Generators.cs

```diff
@@ -330,7 +329,9 @@ private partial class CompilationTracker : ICompilationTracker
                                 identity,
                                 generatedSource.SourceText,
                                 generatedSource.SyntaxTree.Options,
-                                ProjectState.LanguageServices));
+                                ProjectState.LanguageServices,
+                                // Compute the checksum on demand from the given source text.
+                                originalSourceTextChecksum: null));
```

Comment:
this is the code that runs in-proc on the server to actually generate the SG docs.  this is the palce where we want to compute based oin the source text that was generated.